### ISSUE TITLE
allow changing keepalived_ppa_keyserver variable

### ISSUE THREAD

**Issue by jansepke** [evrardjp/ansible-keepalived#48]
I have the problem, that the default hkp port is blocked by our companies firewall. I need to change the current value "keyserver.ubuntu.com" to "hkp://keyserver.ubuntu.com:80". but because the variable is included via include_vars its impossible to change it without deleting it from the vars file.

Could you add another with_first_found item like "keepalived_distribution_vars_file" to the "Gather variables for each operating system" task?

**Comment by evrardjp** [comment]
Hello, 

Sorry for the late answer.

Well this is fairly dangerous to allow an override in the role.

You have other choices for your override:
- Use another source, like UCA
- Override it with your own variable in the cli: -e "keepalived_ppa_keyserver='hkp://keyserver.ubuntu.com:80'" or with -e @<filename>

Yes, I could add a variable file in with first found, but that would require the deployer to set all the variables that are for his distribution, and therefore we'd define "expectations" (deployer would have to follow internal changes in the roles, which sounds problematic).
Alternatively, we could after this "Gather vars for each os" task, have another task to load extra vars if a variable is defined.
But that sounds more a source of problems in the future, compared to the solution above.

**Comment by evrardjp** [comment]
Tell me what you think.

**Comment by jansepke** [comment]
Hi,

thanks for the explanation. I tried UCA and it worked fine. The idea of using extra params in the shell would it make complicated. Another solution: just store `keepalived_ppa_keyserver` in default vars and leave the rest where it is.

**Comment by evrardjp** [comment]
Hey, last question:
Did you use keepalived_ubuntu_src: "uca"  for it?

If not needed, I'll not make a change to prevent stability issues.

**Comment by jansepke** [comment]
yes I used `keepalived_ubuntu_src: "uca"`

**Comment by evrardjp** [comment]
Ok cool, I'll leave this issue closed. If you need anything else, don't hesitate!
